p2p/net/conn/dial.go

```diff
@@ -265,3 +209,100 @@ func MultiaddrNetMatch(tgt ma.Multiaddr, srcs []ma.Multiaddr) ma.Multiaddr {
 	}
 	return nil
 }
+
+type ProtoDialer interface {
```

Comment:
@jbenet I've taken and made the dialers an interface, just like you always wanted!